<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Fouad Yousif, Xihui Lin, Fan Fan, Christopher Lalansingh" />

<meta name="date" content="2017-05-31" />

<title>SeqKat</title>






<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">SeqKat</h1>
<h4 class="author"><em>Fouad Yousif, Xihui Lin, Fan Fan, Christopher Lalansingh</em></h4>
<h4 class="date"><em>2017-05-31</em></h4>



<div id="background" class="section level2">
<h2>Background</h2>
<p>Kataegis is a localized hypermutation occurring when a region is enriched in somatic SNVs (<em>Nik-Zainal S. et al 2012</em>). Kataegis can result from multiple cytosine deaminations catalyzed by the AID/APOBEC family of proteins. A first step to understand kataegis requires the ability to reproducibly and reliability identify it. Although a formal, quantifiable definition of kataegis has not been reached, we have provided the first operational definition in the form of SeqKat, a R package that predicts kataegis from paired tumour normal human whole genome samples. This package contains functions to detect kataegis from SNVs in BED format.</p>
</div>
<div id="approach" class="section level2">
<h2>Approach</h2>
<p>SeqKat uses a sliding window (of fixed width) approach to test deviation of observed SNV trinucleotide content and inter-mutational distance from expected by chance alone. Additionally, an exact binomial test is performed to test that the proportion of each of the 32 tri-nucleotides within each window is higher than expected. The resulting p-values are then adjusted for multiple hypothesis testing using FDR. Hypermutation and kataegic scores are calculated for each window as follows</p>
<blockquote>
<p><em>hypermutation score</em> = <span class="math inline">\(-log_{10}\)</span>(binomial <span class="math inline">\(p_{adj}\)</span>) * <span class="math inline">\(\frac{N observed Mutations}{N expected Mutations}\)</span> [Equation 1]</p>
</blockquote>
<blockquote>
<p><em>kataegis score</em> = <em>hypermutation score</em> * <span class="math inline">\(\frac{N TCX bases}{N expected TCX bases}\)</span> [Equation 2]</p>
</blockquote>
<p>SeqKat reports both hypermutation score and an APOBEC mediated kataegic score along with the start and end position of each detected event. A reference paper will be added upon publication in an upcoming version of this package.</p>
</div>
<div id="input" class="section level2">
<h2>Input</h2>
<p>SeqKat accepts a SNV BED file per patient with the following columns:</p>
<ul>
<li>chromosome</li>
<li>position</li>
<li>reference base</li>
<li>alternate base</li>
</ul>
</div>
<div id="running-seqkat" class="section level2">
<h2>Running SeqKat</h2>
<pre><code>seqkat(sigcutoff = 5,
       mutdistance = 3.2,
       segnum = 4,
       ref.dir = NULL,
       bed.file = &quot;./&quot;,
       output.dir = &quot;./&quot;,
       chromosome = &quot;all&quot;,
       chromosome.length.file = NULL,
       trinucleotide.count.file = NULL
       )</code></pre>
<ul>
<li><p><strong>sigcutoff</strong>: The minimum hypermutation score used to classify the windows in the sliding binomial test as significant windows. The score is calculated following [Equation 1]. <em>Recommended value</em>: 5</p></li>
<li><p><strong>mutdistance</strong>: The maximum intermutational distance allowed for SNVs to be grouped in the same kataegic event. <em>Recommended value</em>: 3.2</p></li>
<li><p><strong>segnum</strong>: Minimum mutation count. The minimum number of mutations required within a cluster to be identified as kataegic. <em>Recommended value</em>: 4</p></li>
<li><p><strong>ref.dir</strong>: Path to a directory containing the reference genome. Each chromosome should have its own .fa file and chromosomes X and Y are named as chr23 and chr24. The fasta files should contain no header.</p></li>
<li><p><strong>bed.file</strong>: Path to the SNV BED file. The BED file should contain the following information: Chromosome, Position, Reference allele, Alternate allele. The file should be named {SAMPLE_NAME}_snvs.bed. Below is an example of a BED file:</p></li>
</ul>
<pre><code>chr4    17185   G   A
chr4    38640   T   C
chr4    52598   C   T
chr4    53102   C   G
chr4    71989   G   A
chr4    91099   C   G
chr4    91139   G   C
chr4    192852  G   C
chr4    201573  G   C
chr4    212498  C   G</code></pre>
<ul>
<li><p><strong>output.dir</strong>: Path to a directory where output will be created</p></li>
<li><p><strong>chromosome</strong>: The chromosome to be analysed. This can be (1, 2, …, 23, 24) or “all” to run sequentially on all chromosomes</p></li>
<li><p><strong>chromosome.length.file</strong> <em>(provided)</em>: A tab separated file containing the lengths of all chromosomes in the reference genome. Below is an example of a chromosome.length.file for hg19:</p></li>
</ul>
<pre><code>&quot;num&quot; &quot;length&quot;
&quot;1&quot; 249250621
&quot;2&quot; 243199373
&quot;3&quot; 198022430
&quot;4&quot; 191154276
&quot;5&quot; 180915260
&quot;6&quot; 171115067
&quot;7&quot; 159138663
&quot;8&quot; 146364022
&quot;9&quot; 141213431
&quot;10&quot; 135534747
&quot;11&quot; 135006516
&quot;12&quot; 133851895
&quot;13&quot; 115169878
&quot;14&quot; 107349540
&quot;15&quot; 102531392
&quot;16&quot; 90354753
&quot;17&quot; 81195210
&quot;18&quot; 78077248
&quot;19&quot; 59128983
&quot;20&quot; 63025520
&quot;21&quot; 48129895
&quot;22&quot; 51304566
&quot;23&quot; 155270560
&quot;24&quot; 59373566
&quot;sum.f&quot; 3036303846
&quot;sum.m&quot; 3095677412</code></pre>
<ul>
<li><strong>trinucleotide.count.file</strong> <em>(provided)</em>: A tab seprarated file containing a count of all trinucleotides present in the reference genome. This can be generated with the get.trinucleotide.counts() function in this package. Below is an example of trinuclotide.count.file for hg19:</li>
</ul>
<pre><code>trinucleotide   count
ACA 118307548
ACC 67377361
ACG 15031779
ACT 94371148
ATA 120046083
ATC 78231773
ATG 107040211
ATT 145343907
CCA 107257513
CCC 75979793
CCG 16160851
CCT 103230644
CTA 75285140
CTC 98529730
CTG 118268845
CTT 117139678
GCA 84247974
GCC 68786498
GCG 13995012
GCT 81425256
GTA 65911575
GTC 54961008
GTG 88082699
GTT 86132552
TCA 114943318
TCC 90485129
TCG 13117247
TCT 130196437
TTA 120231763
TTC 117019750
TTG 111364601
TTT 225211336</code></pre>
<p><strong>note</strong>: sigcutoff, multidistance and segnum default parameters are optimized using <em>Alexandrov et al</em>’s “Signatures of mutational processes in human cancer” dataset.</p>
<p><strong>note</strong>: trinucleotide.count.file and chromosome.length.file have been provided for GRCh38 reference as well</p>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>If Kataegic events are detected, SeqKat generates a tab delimited file that includes details about the detected events. Each line represents one detected hypermutation or kataegic event. The file includes the following columns:</p>
<ul>
<li><strong>sample</strong>: sample name</li>
<li><strong>chr</strong>: chromosome</li>
<li><strong>start</strong>: coordinate indicating where the event starts</li>
<li><strong>end</strong>: coordinate indicating where the event ends</li>
<li><strong>variants</strong>: number of SNVs within the event</li>
<li><strong>score.hm</strong>: hypermutation score.</li>
<li><strong>score.kat</strong>: kataegic score</li>
</ul>
<p><strong>note</strong>: if no event is detected then no file is generated</p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<p>A subset BED file from the publically available breast cancer sample PD4120a is provided in the package. This BED contains 2804 SNVs in the first 2,000,000 bases of chromosome 4. A subset FASTA file and chromosome length file have also been provided for <strong>testing purposes only</strong>.</p>
<pre><code>example.bed.file &lt;- paste0(path.package(&quot;SeqKat&quot;),&quot;/extdata/test/PD4120a-chr4-1-2000000_test_snvs.bed&quot;);
example.ref.dir &lt;- paste0(path.package(&quot;SeqKat&quot;),&quot;/extdata/test/ref/&quot;);
example.chromosome.length.file &lt;- paste0(path.package(&quot;SeqKat&quot;),&quot;/extdata/test/length_hg19_chr_test.txt&quot;);
seqkat(
    5,
    3.2,
    2,
    bed.file = example.bed.file,
    output.dir = &quot;.&quot;,
    chromosome = &quot;4&quot;,
    ref.dir = example.ref.dir,
    chromosome.length.file = example.chromosome.length.file
    );</code></pre>
<p>To view the detected events, you can check the file <em>PD4120a-chr4-1-2000000_chr4_cutoff5_mutdist3.2_segnum2.txt</em></p>
<pre><code>sample  chr start   end variants    score.hm    score.kat
PD4120a-chr4-1-2000000  4   1009070 1009541 4   119920.029973896    0</code></pre>
<p>In this example, SeqKat detected one hypermutation window on chromosome 4 between 1009070 and 1009541, containing 4 SNVs with a hypermutation score of 119920.03. The kataegic score is 0, indicating that it is not an APOBEC mediated event.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<ul>
<li><p>Nik-Zainal S, Alexandrov LB, Wedge DC, Van Loo P, Greenman CD, Raine K, Jones D, Hinton J, Marshall J, Stebbings LA, Menzies A, Martin S, Leung K, Chen L, Leroy C, Ramakrishna M, Rance R, Lau KW, Mudie LJ, Varela I, McBride DJ, Bignell GR, Cooke SL, Shlien A, Gamble J, Whitmore I, Maddison M, Tarpey PS, Davies HR, Papaemmanuil E, Stephens PJ, McLaren S, Butler AP, Teague JW, Jönsson G, Garber JE, Silver D, Miron P, Fatima A, Boyault S, Langerød A, Tutt A, Martens JW, Aparicio SA, Borg Å, Salomon AV, Thomas G, Børresen-Dale AL, Richardson AL, Neuberger MS, Futreal PA, Campbell PJ, Stratton MR; Breast Cancer Working Group of the International Cancer Genome Consortium.. <strong>Mutational processes molding the genomes of 21 breast cancers</strong>. <em>Cell</em>. 2012 May 25;149(5):979-93. doi: 10.1016/j.cell.2012.04.024. Epub 2012 May 17. PubMed PMID: 22608084; PubMed Central PMCID: PMC3414841.</p></li>
<li><p>Alexandrov LB, Nik-Zainal S, Wedge DC, Aparicio SA, Behjati S, Biankin AV, Bignell GR, Bolli N, Borg A, Børresen-Dale AL, Boyault S, Burkhardt B, Butler AP, Caldas C, Davies HR, Desmedt C, Eils R, Eyfjörd JE, Foekens JA, Greaves M, Hosoda F, Hutter B, Ilicic T, Imbeaud S, Imielinski M, Jäger N, Jones DT, Jones D, Knappskog S, Kool M, Lakhani SR, López-Otín C, Martin S, Munshi NC, Nakamura H, Northcott PA, Pajic M, Papaemmanuil E, Paradiso A, Pearson JV, Puente XS, Raine K, Ramakrishna M, Richardson AL, Richter J, Rosenstiel P, Schlesner M, Schumacher TN, Span PN, Teague JW, Totoki Y, Tutt AN, Valdés-Mas R, van Buuren MM, van ’t Veer L, Vincent-Salomon A, Waddell N, Yates LR; Australian Pancreatic Cancer Genome Initiative.; ICGC Breast Cancer Consortium.; ICGC MMML-Seq Consortium.; ICGC PedBrain., Zucman-Rossi J, Futreal PA, McDermott U, Lichter P, Meyerson M, Grimmond SM, Siebert R, Campo E, Shibata T, Pfister SM, Campbell PJ, Stratton MR. <strong>Signatures of mutational processes in human cancer</strong>. <em>Nature</em>. 2013 Aug 22;500(7463):415-21. doi: 10.1038/nature12477. Epub 2013 Aug 14. Erratum in: Nature. 2013 Oct 10;502(7470):258. Imielinsk, Marcin [corrected to Imielinski, Marcin]. PubMed PMID: 23945592; PubMed Central PMCID: PMC3776390.</p></li>
</ul>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
